modules/web/web-api/src/main/java/com/enonic/xp/web/servlet/ServletRequestUrlHelper.java

```diff
@@ -126,13 +126,24 @@ public static String getServerUrl()
     }
 
     public static String getServerUrl( final HttpServletRequest req )
+    {
+        return getServerUrl( req, false );
+    }
+
+    public static String getServerUrl( final HttpServletRequest req, final boolean websocket )
     {
         final StringBuilder str = new StringBuilder();
 
         //Appends the scheme part
         String scheme = req.getScheme();
-        str.append( scheme );
-
+        if ( websocket )
```

Comment:
This entire class deserves depreciation. Don't extend it

Reply:
Any suggestions? This is a public API and that should be done here for now. I am ready to discuss this issue if needed.